src/Uno.Analyzers/UnoInitializeComponentAnalyzer.cs

```diff
@@ -0,0 +1,72 @@
+﻿#nullable enable
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Uno.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class UnoInitializeComponentAnalyzer : DiagnosticAnalyzer
+{
+	internal const string Title = "Call 'InitializeComponent()' from code-behind";
+	internal const string MessageFormat = "Type '{0}' should call 'InitializeComponent()' from its constructor";
+	internal const string Category = "Correctness";
+
+	internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+#pragma warning disable RS2008 // Enable analyzer release tracking
+		"Uno0006",
+#pragma warning restore RS2008 // Enable analyzer release tracking
+		Title,
+		MessageFormat,
+		Category,
+		DiagnosticSeverity.Warning,
+		isEnabledByDefault: true,
+		helpLinkUri: "https://aka.platform.uno/UNO0006"
+	);
+
+	public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+	public override void Initialize(AnalysisContext context)
+	{
+		context.EnableConcurrentExecution();
+		context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+		context.RegisterSymbolStartAction(context =>
+		{
+			var type = (INamedTypeSymbol)context.Symbol;
+			IMethodSymbol? initializeComponent = null;
+			foreach (var member in type.GetMembers("InitializeComponent"))
+			{
+				if (member is IMethodSymbol { Parameters.Length: 0 } method)
```

Comment:
Yes, analyzers should be running on the updated compilation.

Generators will run here: https://github.com/dotnet/roslyn/blob/e1e7d07474a64f4da5c8b0b7f57082de5ce6a22b/src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs#L1146, and the resulting compilation is re-assigned to `compilation`.

Then analyzers run later:

https://github.com/dotnet/roslyn/blob/e1e7d07474a64f4da5c8b0b7f57082de5ce6a22b/src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs#L1213

Reply:
Confirmed it works.

![image](https://github.com/unoplatform/uno/assets/31348972/845f6a76-c4b1-44a0-81e6-b99794dda7c6)

Reply:
Ah yes, that's true! Then in that case, we probably should validating that this is coming from a generated file? Also, let's limit this rule to types inheriting from UIElement.

Reply:
`Application` isn't a UIElement, and ResourceDictionary as well. I feel it should be safe as `InitializeComponent` isn't a common name besides what we really want to look at?

Reply:
Let's make explicit cases for those then. Maybe `DependencyObject` is a better choice, also. The ideal solution would be to find XAML files and find x:Class that matches. This way, we'd exactly know if the type is eligible, but it would be more expensive...

Reply:
Yup, I wanted to try to make it as performant as possible. But I'm also okay with the slightly less performant approach. It was actually the first thing I tried (first commit in PR). Assuming it doesn't end up having to be a compilation-end analyzer, I have no strong opinion.